sdk/nodejs/runtime/closure/serializeClosure.ts

```diff
@@ -144,8 +144,17 @@ function serializeJavaScriptText(
     // Export the appropriate value.  For a normal function, this will just be exporting the name of
     // the module function we created by serializing it.  For a factory function this will export
     // the function produced by invoking the factory function once.
-    const text = `exports.${exportName} = ${outerFunctionName}${isFactoryFunction ? "()" : ""};\n`
-        + environmentText + functionText;
+    let text: string;
+    if (isFactoryFunction) {
+        // for a factory function, we need to call the function at the end.  That way all the logic
+        // to set up the environment has run.
+        text = environmentText + functionText + "\n" +
+            `exports.${exportName} = ${outerFunctionName}${isFactoryFunction ? "()" : ""};`;
+    }
+    else {
+        text = `exports.${exportName} = ${outerFunctionName}${isFactoryFunction ? "()" : ""};\n`
+            + environmentText + functionText;
+    }
```

Comment:
@swgillespie This is what i was discussing with you.  I could make thsi simpler and only have the top codepath run.  But htat would then involve updating all tests.  I'm totally willing ot do that if you think it's worth it.  But i'm just being lazy here...